pkg/kubelet/kubelet.go

```diff
@@ -2176,6 +2187,9 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 			// TODO: Do we want to support this?
 			glog.Errorf("Kubelet does not support snapshot update")
 		}
+
+		kl.sourcesReady.AddSource(u.Source)
```

Comment:
@jingxu97 @saad-ali 
If it is important to move this after the handlers are called do we want to comment why? Can we protect it with code?